packages/storage/amplify_storage_s3_ios/ios/Classes/SwiftStorageS3.swift

```diff
@@ -21,62 +21,69 @@ import AWSMobileClient
 import amplify_core
 
 public class SwiftStorageS3: NSObject, FlutterPlugin {
+    private var channel: FlutterMethodChannel?
     
     private static var transferProgressStreamHandler : TransferProgressStreamHandler = TransferProgressStreamHandler()
     
     public static func register(with registrar: FlutterPluginRegistrar) {
-        let channel = FlutterMethodChannel(name: "com.amazonaws.amplify/storage_s3", binaryMessenger: registrar.messenger())
         let instance = SwiftStorageS3()
-        registrar.addMethodCallDelegate(instance, channel: channel)
+        instance.channel = FlutterMethodChannel(name: "com.amazonaws.amplify/storage_s3", binaryMessenger: registrar.messenger())
+        registrar.addMethodCallDelegate(instance, channel: instance.channel!)
         
         FlutterEventChannel(name: "com.amazonaws.amplify/storage_transfer_progress_events", binaryMessenger: registrar.messenger()).setStreamHandler(transferProgressStreamHandler)
     }
+    
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        let result = AtomicResult(result, call.method)
+        let arguments = call.arguments as! Dictionary<String,AnyObject>
 
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    let result = AtomicResult(result, call.method)
-
-    if(call.method == "addPlugin"){
-        do {
-            try Amplify.add(plugin: AWSS3StoragePlugin() )
-            result(true)
-        } catch let error{
-            if(error is StorageError){
-                let storageError = error as! StorageError
-
-                ErrorUtil.postErrorToFlutterChannel(
-                    result: result,
-                    errorCode: "StorageException",
-                    details: [
-                        "message" : storageError.errorDescription,
-                        "recoverySuggestion" : storageError.recoverySuggestion,
-                        "underlyingError": storageError.underlyingError != nil ? storageError.underlyingError!.localizedDescription : ""
-                    ])
-            } else if(error is ConfigurationError) {
-                let configError = error as! ConfigurationError
+        switch call.method {
+        case "configureStorage":
+            do {
+                guard let hasPrefixResolver = arguments["hasPrefixResolver"] as? Bool else{
+                    _ = false
+                    return
+                }
```

Comment:
Sorry my original comment may be false. but this early return seems incorrect here... as it's not invoking `result(true)`.
How about

```suggestion
                let hasPrefixResolver = arguments["hasPrefixResolver"] as? Bool ?? false
```

Reply:
Thanks Hui, code has been updated.